### PR TITLE
Extensible RUM events for data retention

### PR DESCRIPTION
**Pull request by rtrieu**
### What does this PR do?
Adds a clarifying note about data retention for specific RUM events.

### Motivation
Helpful detail for customers [DOCS-5588](https://datadoghq.atlassian.net/browse/DOCS-5588)

### Additional Notes
Ready for merge following a review from @keithbradshaw or anyone from @DataDog/rum-app 

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-5588]: https://datadoghq.atlassian.net/browse/DOCS-5588?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/developers/guide/data-collection-resolution-retention.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Find below a summary of Datadog data collection, resolution, and retention:
...
 |Logs                |Logs                                          |Datadog Agent + Logs, third-party log collectors, or API                                        |Real time            |1 millisecond        |Plan                                                                                        |
-|Real User Monitoring|Error Tracking                                |RUM SDK                                                                                       |2 minutes            |60 seconds           |30 days                                                                                     |
+|Real User Monitoring|Error Tracking                                |RUM SDK                                                                                       |2 minutes            |60 seconds           |30 days <br/>View, Error, and User Action events can be extended to 90 days at no cost by reaching out to support                                                                                     |
```
Comment (alai97) [inline content/en/developers/guide/data-collection-resolution-retention.md:47]:
```suggestion
|Real User Monitoring|Error Tracking                                |RUM SDK                                                                                       |2 minutes            |60 seconds           |30 days <br/>View, Error, and User Action events can be extended to 90 days at no cost by reaching out to Support.                                                                                     |
```

File: content/en/developers/guide/data-collection-resolution-retention.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ Find below a summary of Datadog data collection, resolution, and retention:
...
 |Real User Monitoring|Real User Monitoring                          |RUM SDK                                                                                       |Real time            |1 millisecond        |30 days for session, view, action and error events <br/>15 days for resource and long task events|
```
Comment (alai97) [inline content/en/developers/guide/data-collection-resolution-retention.md:48]:
```suggestion
|Real User Monitoring|Real User Monitoring                          |RUM SDK                                                                                       |Real time            |1 millisecond        |30 days for session, view, action, and error events <br/>15 days for resource and long task events|
```

**Review by alai97 — APPROVED** [review]
Looks good, some suggestions!

**Review by anediaz — APPROVED** [review]
LGTM for RUM-app

**Pull request merged**